src/components/sections/general/form-sections/sync-checkbox-field.js

```diff
@@ -0,0 +1,26 @@
+import React from 'react'
+import PropTypes from '@dhis2/prop-types'
+import { CheckboxField } from '@dhis2/ui'
+import { FormSection } from './form-section'
+
+export const SyncCheckboxField = ({ handleForm, syncElement }) => {
+    const { label, helpText, syncType } = syncElement
+    return (
+        <FormSection>
+            <CheckboxField
+                label={label}
+                helpText={helpText}
+                {...handleForm.getCheckbox(syncType)}
```

Comment:
The caller needs to expand this `getCheckbox` result, otherwise it will re-render every time (because a new object is returned)

Ideally this should specify exactly the properties the component accepts  in PropTypes:

```
name,
checked
disabled
type
onChange
```

Though `type` is ALWAYS `checkbox` so probably doesn't need to be a prop.

Reply:
I'm not sure if I got it, but if by expanding getCheckbox means this:

```
<SyncCheckboxField
           label={generalSettingsFormSections.encryptDB.label}
           helpText={generalSettingsFormSections.encryptDB.helpText}
           name={generalSettingsFormSections.encryptDB.syncType}
           checked={handleForm.fields[generalSettingsFormSections.encryptDB.syncType]}
           disabled={handleForm.fields.disableAll}
           onChange={handleForm.handleEncryptDialog.onChange}
/>


export const SyncCheckboxField = ({ label, helpText, name, checked, disabled, onChange }) => (
        <FormSection>
            <CheckboxField
                label={label}
                helpText={helpText}
                name={name}
                checked={checked}
                disabled={disabled}
                type="checkbox"
                onChange={onChange}
            />
        </FormSection>
)
```

the same idea should apply to the other components right?